middleware/src/middleware.go

```diff
@@ -251,6 +256,37 @@ func (middleware *Middleware) Restore(method rpcmessages.RestoreArgs) (rpcmessag
 	}
 }
 
+// Hostname returns a GenericResponse struct in response to a rpcserver request
+func (middleware *Middleware) Hostname(args rpcmessages.HostnameArgs) (rpcmessages.GenericResponse, error) {
+	switch args.Method {
```

Comment:
@tblock about `GenericResponse`. I see that it is easier to use when coding, but i would reconsider and start using descriptive structs (and methods!) for everything, as you don't want to deal with a lot of implicit assumptions when reading/maintaining that code.

for example, in this call:

if method==get,genericresponse.Message is the hostname.
if method==set,  what is the Message? need to look at the code, not sure what it is in case of success...


More importantly: we have a great mechanism to add rpc methods, and in this one, you inlining a 2nd layer rpc protocol into one rpc method. This only complicates the code and you need to handle more state (e.g. the default case below). 

Better to just have two mthods: `GetHostname` and `SetHostname` calls, with explicit struct fields.

Reply:
> I see that it is easier to use when coding, but i would reconsider and start using descriptive structs (and methods!) for everything, as you don't want to deal with a lot of implicit assumptions when reading/maintaining that code.

> Better to just have two methods: `GetHostname` and `SetHostname` calls, with explicit input and output structs. 

I agree.